src/js/src/mixins/SearchUtils.js

```diff
@@ -0,0 +1,103 @@
+             testString.substring(0,10) === 'url_norm:"'
+    },
+    //Deliver a normal search
+    deliverSearchRequest(futureQuery, updateHistory) {
+      this.requestSearch({query:futureQuery, facets:this.searchAppliedFacets, options:this.solrSettings})
+      this.requestFacets({query:futureQuery, facets:this.searchAppliedFacets, options:this.solrSettings})
+      updateHistory ? this.$_pushSearchHistory('SolrWayback', futureQuery, this.searchAppliedFacets, this.solrSettings) : null
+    },
+    //Deliver an URL search
+    deliverUrlSearchRequest(futureQuery, updateHistory) {
+      this.updatePreNormalizedQuery(futureQuery)
+      if(this.$_validateUrlSearchPrefix(this.DisectQueryForNewUrlSearch(futureQuery))) {
+        this.requestUrlSearch({query:this.DisectQueryForNewUrlSearch(futureQuery), facets:this.searchAppliedFacets, options:this.solrSettings})
+        this.requestFacets({query:'url_norm:"' + this.DisectQueryForNewUrlSearch(futureQuery) + '"', facets:this.searchAppliedFacets, options:this.solrSettings})
+        updateHistory ? this.$_pushSearchHistory('SolrWayback', this.DisectQueryForNewUrlSearch(futureQuery), this.searchAppliedFacets, this.solrSettings) : null
+      }
+      else {
+        this.setNotification({
+          title: 'We are so sorry!',
+          text: 'This URL is not valid. the url must start with \'http://\' or \'https://\'',
+          type: 'error',
+          timeout: false
+        })
+      }
+    },
+    // Deliver an image search
+    deliverImgSearchRequest(futureQuery, updateHistory) {
+      this.requestImageSearch({query:futureQuery})
+      updateHistory ? this.$_pushSearchHistory('SolrWayback', futureQuery, this.searchAppliedFacets, this.solrSettings) : null
+    },
+    // Check if there has been any changes to the query
+    queryHasChanged(query) {
+      return query !== this.query
+    },
+    // Prepare for a new search
+    prepareVariablesForNewSearch(futureQuery) {
+      this.updatePreNormalizedQuery(null)
+      this.clearResults()
+      this.updateQuery(futureQuery)
+    },
+    // Disect the query for URL searching
+    DisectQueryForNewUrlSearch(futureQuery) {
+      let queryString = ''
+          if(futureQuery.substring(0,10) === 'url_norm:"') {
+            queryString = futureQuery.replace('url_norm:"', '')
+            queryString.substring(queryString.length-1, queryString.length) === '"' ? queryString = queryString.slice(0,-1) : null
+          }
+          else {
+            queryString = futureQuery
+          }
+          return queryString
+    },
+    // Method to fire off a search (and deciding which kind it is)
+    $_determineNewSearch(futureQuery, updateHistory) {
+      //console.log('we\'ve accessed the searchfunction with ',futureQuery)
+      //console.log('we have these solrsettings: ', this.solrSettings)
+      //console.log('and these facets', this.searchAppliedFacets)
+      this.prepareVariablesForNewSearch(futureQuery)
```

Comment:
prepareStateForNewSearch would be a better name here

Reply:
:+1: